decisions/websre/0002-web-sre-service-documentation.md

```diff
@@ -0,0 +1,48 @@
+# 2. Web SRE Service Documentation
+
+Date: 2021-06-01
+Scope: Web SRE Team
+
+## Status
+
+Proposed
+
+## Context
+
+The Web SRE Team manages a number of services that vary in users, technologies used, setup, our commitment, infrastructure, process, etc. 
+
+Pre-existing documentation for these services, if it exists, also varies - in location, level of detail, scope, structure, presumed audience, etc.
+
+To help share knowledge between the Web SRE Team about these services' unique situations, a consistent baseline for our services documentation is a required first step.
+
+## Decision
+
+A Web SRE service here is defined as any codebase where we have some ownership over an aspect of this codebase being functional - e.g. the deployment process, automation, infrastructure running the service, observability of some aspect(s), or authoring the code ourselves. Web SRE services could be externally or internally-facing, including infrastructure services only Web SRE are aware of and leverage.
+
+For each Web SRE Service, we will author a Service Documentation page in Mana as a child of https://mana.mozilla.org/wiki/display/SRE/Service+Documentation.
+
+Service Documentation pages:
+* use the Web SRE Services Docs confluence page template for its structure;
+* have a primary audience of the Web SRE Team itself, with a secondary audience of other SRE teams;
+* are open to Mozilla internally, however not all teams of Mozilla are the intended audience of these pages;
+* have Runbook pages as child pages to the relevant Service Documentation page;
+* are maintained to reflect the current state - to the best of our knowledge - of a service's context, including a service being decommissioned (the Service Documentation page should note that but be left up as a tombstone marker of the decision);
+* cover Web SRE Services as defined above.
+
+Service Documentation pages do not:
+* include all possible services Web SRE might work with - e.g. we don't document AWS Services generically or services owned and maintained by other teams; 
```

Comment:
I think the existing do-not item about repeating things is great.  (don't document aws/don't try to be for all audiences/don't include all the service documentation).   I wonder if something like, 'we prefer to link to good sources of documentation when possible, over duplicating them or rewording into our own voice' would be good to add.

Though for runbooks specifically I'd prefer that we copy the info into the runbook, so if the link were to change structure or content you don't have to deal with that at 3am.  (That seems out of scope for this adr though).

Reply:
(I'm wondering if if the second part of my comment should turn into dos for runbooks and howto pages).  Apologies for these very much being in my voice, but I figured it'd be easier to work.  Curious of if you think these are in scope for our current adr and/or are they useful in any adr?

Runbook pages:
* Should include ways to validate what state a system is in (how to reproduce the problem)
* Should explain what problem the symptom indicates
* Should give a very opinionated way to step through the problem
* Should be as self contained as possible (redirect to other pages rarely, and external pages even more rarely, even at the risk of repeating information)
* Should give commands to resolve the problem as clearly as possible
* should outline a fallback plan (who to call, what to do next)


How-to pages:
* Should repeat exist documentation as little as possible.  Linking to external docs is encouraged.  To be augmented with our specific contexts.
* Should assume a high level of competence from the audience (don't explain how to download a csv file, though giving an example command to establish context is great)
* Should explain the decision points in a process, and how to make them
* Should outline what is needed to perform the work as early as possible in the document (what access, what tools)
* Should outline who needs to approve the work, or how to decide if it's safe to do the work as early as possible

Reply:
hmm let me try this out and see how it looks, thanks for writing it up

Reply:
added a form of your additions to the doc, with some modifications and transition. let me know your thoughts.